### PR TITLE
feat: generate the import declaration for the completion item code ac…

### PR DESCRIPTION
**Pull request by ivanwonder**
…tions

In the completion item, the `additionalTextEdits` can only be included the changes about the current file, the other changes should be inserted by the vscode command.

For example, when the user selects a component in an HTML file, the extension inserts the selector in the HTML file and auto-generates the import declaration in the TS file.

### CODE REVIEW

**Comment by ivanwonder** [comment]
Blocked by [PR](https://github.com/angular/angular/pull/55595)

**Review by dylhunn — APPROVED** [review]

**Comment by dylhunn** [comment]
@ivanwonder Looks like the CI is failing, due to the new test:
```
Error: Recevied parameters by name but param is not an object literal.
```

File: integration/lsp/ivy_spec.ts [hunk 1]
```diff
@@ -580,6 +580,58 @@ export class AppComponent {
...
+      }) as lsp.CompletionItem[];
+      const libPostResponse = response.find(res => res.label === 'baz-component')!;
+      const detail = await client.sendRequest(lsp.CompletionResolveRequest.type, libPostResponse);
+      expect(detail.additionalTextEdits).toEqual([{
+        'newText':
+            '{\n    selector: \'bar-component\',\n    template: `<`,\n    standalone: true,\n    imports: [BazComponent]\n}',
+        'range': {'start': {'line': 11, 'character': 11}, 'end': {'line': 15, 'character': 1}}
+      }]);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (ivanwonder) [reply integration/lsp/ivy_spec.ts:632]: This PR just reuses the work of code action, with no bug fix. I will put it on my to-do list and look into it.

**Comment by ivanwonder** [comment]
https://github.com/angular/vscode-ng-language-service/assets/15815660/02382282-8c3e-4a71-815b-67e7966d5e65

https://github.com/angular/vscode-ng-language-service/assets/15815660/e8879515-4f0e-4738-b360-bf87da6095b8

Now the format is not perfect. For example, when the `imports` array has multi-line, the indentation of the identifier in the array is wrong because I only print the `PropertyAssignment` node, and the indentation of the object is lost. We can optimize it later.

image

**Comment by ivanwonder** [comment]
> error @angular/language-service@18.1.0-next.2: The engine "node" is incompatible with this module. Expected version "^18.19.1 || ^20.11.1 || >=22.0.0". Got "18.13.0"

@dylhunn The node in the docker needs to be updated

File: pnpm-lock.yaml
Comment (ivanwonder) [inline pnpm-lock.yaml:1]: There are lots of diffs here. I don't know whether it's acceptable.

**Review by dylhunn — APPROVED** [review]

**Comment by dylhunn** [comment]
The substance of the change looks good. But I'm also not sure why changing the language-service version has generated the changes to pnpm lock. @atscott Do you think this is expected?

**Comment by dylhunn** [comment]
caretaker: do not merge until we get the pnpm lock issue sorted out.

File: pnpm-lock.yaml
Comment (atscott) [reply pnpm-lock.yaml:1]: This happens if you use `yarn bazel run...` to update the files. You have to instead use global bazel for `bazel run @npm//:sync`

**Comment by atscott** [comment]
> changes to pnpm lock. @atscott Do you know if this is expected?

Yea, this happens when doing `yarn bazel run @npm//:sync`. You have to instead use global bazel `bazel run @npm//:sync` to do it...

File: pnpm-lock.yaml
Comment (ivanwonder) [reply pnpm-lock.yaml:1]: Done, but there is a new error.

I think this is not caused by my code.

```
Writing SSH key for checkout to "/home/circleci/.ssh/id_rsa"
Writing SSH public key for checkout to "/home/circleci/.ssh/id_rsa.pub"
Cloning git repository
Fetching from remote repository
Checking out branch
Cloning into '.'...
From github.com:angular/vscode-ng-language-service
 * [new ref]             refs/pull/2031/head -> origin/pull/2031
fatal: reference is not a tree: 33b10d17b795f19c4bf7f38e9079c2195a220262
failed to checkout: exit status 128

failed to checkout: exit status 128
```

**Comment by dylhunn** [comment]
@ivanwonder Is this ready for merge, or do you want Andrew's review also? If so, I'll tag it to be merged

**Comment by ivanwonder** [comment]
> @ivanwonder Is this ready for merge, or do you want Andrew's review also? If so, I'll tag it to be merged

Yes, It's ready. It would be better if Andrew had time to take a look.

**Review by atscott — APPROVED** [review]
Really cool! Great work!

**Pull request merged**